darts/tests/models/forecasting/test_torch_forecasting_model.py

```diff
@@ -2515,6 +2517,156 @@ def test_predict_after_fit_from_dataset(self):
+            # Case A: Darts' automatic checkpointing
+            kwargs_last = dict(
+                common_kwargs, model_name="last_model", save_checkpoints=True
+            )
+            kwargs_best = dict(
+                common_kwargs, model_name="best_model", save_checkpoints=True
+            )
+        else:
+            # Case B: User provides a custom `trainer` with a ModelCheckpoint callback
+            kwargs_last = dict(
+                common_kwargs, model_name="last_model", save_checkpoints=False
+            )
+            kwargs_best = dict(
+                common_kwargs, model_name="best_model", save_checkpoints=False
+            )
+
+            # for last model
+            checkpoints_folder_last = os.path.join(tmpdir_fn, "last_checkpoints")
+            checkpoint_callback_last = ModelCheckpoint(
+                dirpath=checkpoints_folder_last,
+                save_last=True,
+                monitor="val_loss",
+                filename="best-{epoch}-{val_loss:.2f}",
+            )
+            trainer_last = pl.Trainer(
+                max_epochs=n_epochs,
+                callbacks=[checkpoint_callback_last],
+                **tfm_kwargs["pl_trainer_kwargs"],
+            )
+
+            # for best model
+            checkpoints_folder_best = os.path.join(tmpdir_fn, "best_checkpoints")
+            checkpoint_callback_best = ModelCheckpoint(
+                dirpath=checkpoints_folder_best,
+                save_last=True,
+                monitor="val_loss",
+                filename="best-{epoch}-{val_loss:.2f}",
+            )
+            trainer_best = pl.Trainer(
+                max_epochs=n_epochs,
+                callbacks=[checkpoint_callback_best],
+                **tfm_kwargs["pl_trainer_kwargs"],
+            )
+
+        model_last = RNNModel(**kwargs_last)
+        model_last.fit(
+            series_train,
+            val_series=series_val,
+            load_best=False,
+            trainer=trainer_last,
+        )
+        last_model_epochs_trained = model_last.epochs_trained
+
+        model_best = RNNModel(**kwargs_best)
+        model_best.fit(
+            series_train, val_series=series_val, load_best=True, trainer=trainer_best
+        )
+        best_model_epochs_trained = model_best.epochs_trained
+
+        # The model trained to the end should have n_epochs trained
+        assert last_model_epochs_trained == n_epochs
+        # The model loading the best checkpoint should have trained fewer epochs (or equal if best is last)
+        # With our crafted data, it's highly likely to be less.
+        assert best_model_epochs_trained < last_model_epochs_trained
+
+        # We can also check that the trainer object confirms the best model was loaded
+        trainer_to_check = trainer_best if use_custom_trainer else model_best.trainer
+        best_model_path = trainer_to_check.checkpoint_callback.best_model_path
+        assert best_model_path and os.path.exists(best_model_path)
```

Comment:
Nice :) And now I as a last thing we should check that loading the best model checkpoint from `model_last` gives identical predicitons to `model_best`.

E.g. something like below


```py
model_last_best = RNNModel.load_from_checkpoint(
    model_name=model_last.model_name,
    work_dir=model_last.work_dir,
    best=True,
)

preds_last_best = model_last_best.predict(n=1)
assert preds_last_best == preds_best
```